app/src/main/java/com/codingblocks/cbonlineapp/course/CourseViewModel.kt

```diff
@@ -41,7 +47,10 @@ class CourseViewModel(
                 is ResultWrapper.GenericError -> setError(response.error)
                 is ResultWrapper.Success -> with(response.value) {
                     if (isSuccessful) {
-                        course.postValue(body())
+                        if (prefs.SP_JWT_TOKEN_KEY.isNotEmpty())
+                            checkIfCourseWishlisted(body())
+                        else
+                            course.postValue(body())
```

Comment:
why this depends upon the user is logged in or not it shouldn't depend upon user login no matter what